app/assets/javascripts/annotot/annotot_endpoint.js

```diff
@@ -158,13 +158,15 @@
     // Converts OA Annotation to endpoint format
     getAnnotationInEndpoint: function(oaAnnotation) {
       // Generate a new uuid if one is not present
-      if (oaAnnotation['@id'] == null) {
+      // condition needs to be set to "undefined" to catch instances without an @id
+      if (oaAnnotation['@id'] === undefined) {
```

Comment:
I think we may also need to check for `null` here as it could be "undefined" but still null. Right?

```javascript
var test = {}
test["@id"] === undefined
> true
test["@id"] === null
> false

var test = {"@id": null}
test["@id"] === undefined
> false
test["@id"] === null
> true
```

Reply:
This should give us the desired behavior:

`!oaAnnotation['@id']`

```javascript
var oaAnnotation = {};
console.log("{} ", !oaAnnotation["@id"]);

var oaAnnotation = { "@id": null };
console.log("{'@id': null}", !oaAnnotation["@id"]);

var oaAnnotation = { "@id": "test" };
console.log("{'@id': 'test'}", !oaAnnotation["@id"]);
```